diff -crB --new-file p1/rb/config/Make.rules.Darwin p2/rb/config/Make.rules.Darwin
*** p1/rb/config/Make.rules.Darwin	1969-12-31 16:00:00.000000000 -0800
--- p2/rb/config/Make.rules.Darwin	2013-08-18 13:55:30.000000000 -0700
***************
*** 0 ****
--- 1,18 ----
+ mklibfilename           = $(if $(2),$(1).$(2).dylib,$(1).dylib)
+ mksoname                = $(if $(2),$(1).$(2).dylib,$(1).dylib)
+ 
+ mklibname               = $(1).dylib
+ curdir                  = $(shell pwd)
+ 
+ ifeq ($(OPTIMIZE),yes)
+    ifneq ($(embedded_runpath_prefix),)
+       mkshlib                 = $(CXX)  -dynamiclib $(shlibldflags) -o $(1) -install_name $(runpath_libdir)/$(2) $(3) $(4)
+    else
+       mkshlib                 = $(CXX)  -dynamiclib $(shlibldflags) -o $(1) -install_name $(2) $(3) $(4)
+    endif
+ else
+     # Don't use -install_name if we're building a debug version. GDB doesn't honor the
+     # DYLD_LIBRARY_PATH environment variable if a runpath is embedded in the
+     # executables or libraries.
+     mkshlib                 = $(CXX)  -dynamiclib $(shlibldflags) -o $(1) $(3) $(4)
+ endif
